tests/wallet/nft_wallet/test_nft_wallet.py

```diff
@@ -33,13 +33,43 @@ async def tx_in_pool(mempool: MempoolManager, tx_id: bytes32) -> bool:
     return True
 
 
+def assert_rpc_response(response, condition_func=None) -> Dict:
+    assert isinstance(response, dict)
+    assert response.get("success")
+    if condition_func:
+        assert condition_func(response)
+    return response
+
+
+async def wait_rpc_state_condition(timeout, coroutine, params, condition_func) -> Dict:
+    while timeout > 0:
+        resp = await coroutine(*params)
+        if condition_func(resp):
+            assert isinstance(resp, dict)
+            return resp
+        await asyncio.sleep(0.5)
+        timeout -= 0.5
```

Comment:
This drifts.  It should be of the form of identifying the termination time (use `time.monotonic()` not `time.time()`) and just looping until then.  No timeout decrement.

It's not ready yet, but looking to the future there will be https://github.com/Chia-Network/chia-blockchain/pull/11535/files#diff-ed857c49736dea1b775dddce47f5ec9989dd7525d3558c656db792a3c42c9830R22-R37 and probably some layers around it.